From 2834315b0673c1dbeb2014dea57095ccf43e8d88 Mon Sep 17 00:00:00 2001
From: thampan <joby.thampan@intel.com>
Date: Wed, 24 Jul 2019 14:46:19 +0800
Subject: [PATCH] PONRTSYS-4764: From PMAC HW cannot do padding, Since along
 with Padding HW will regenerate FCS, Reverting the change done in GSWIP
 driver

---
 drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c b/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
index fb5c3c8b9ff6..8725428602a3 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
@@ -539,14 +539,6 @@ static int pmac_glbl_cfg(struct core_ops *ops, u8 pmacid)
 	glbl_cfg.bProcFlagsEgCfgEna = 1;
 	glbl_cfg.eProcFlagsEgCfg = GSW_PMAC_PROC_FLAGS_MIX;
 
-	/* For PON HGU, ARP or other small packet size generated from
-	 * CPU -> PONIP -> OLT -> STC is getting droped at OLT, due to underize.
-	 * Adding padding to all packets less than 64 bytes to PON IP at PMAC
-	 * level. Currently all packets to PON IP is through PMAC 1
-	 */
-	if (pmacid == PMAC_1)
-		glbl_cfg.bPadEna = 1;
-
 	ops->gsw_pmac_ops.Pmac_Gbl_CfgSet(ops, &glbl_cfg);
 
 	return 0;
